canonical/prometheus-k8s-operator
# Implement grafana_datasource_exchange

## Issue
Add a `send-datasource` endpoint to exchange datasource UIDs with other COS components.

## Context
TODO:
- [x] merge `grafana_source` lib https://github.com/canonical/grafana-k8s-operator/pull/364

## Testing Instructions
1. Deploy Tempo HA
2. Deploy `grafana-k8s`
3. Deploy `prometheus-k8s` from this branch
4. Integrate `tempo` and `grafana` over `grafana_datasource`
5. Integrate `prometheus` and `grafana` over `grafana_datasource`
6. Integrate `prometheus` and `tempo` over `grafana_datasource_exchange`
7. `jhack show-relation prometheus tempo` and observe the datasource UID of `prometheus` shared to `tempo` 

## Drive-bys
- Use `ops[testing]` and refactor scenario tests
- remove unneeded tox deps
